### DISCUSSION TITLE
Select uncommon commands from a list

### DISCUSSION THREAD

**Discussion by user202729** [opensteno/plover#1323] [Ideas]
So this is an idea...

For example, occasionally I need to take a screenshot (that currently I map to `skree*-sh`, with the plover-run-shell plugin).

But I only very rarely need to do that, so usually I would have forgotten which stroke I assigned to do it.

A possible approach would be to make a theory to represent it correctly (such as (some prefix stroke to denote run a new program) + (first syllable of the program name)). (*I didn't try this approach*)

An alternative that I've been using for a similar task (enter Unicode characters) is to map `TKPH-PL` (think nk-m for Unicode-math) to `{PLOVER:SHELL: python -m picker.rofimoji -f math &}{^}`, then choose the entry.

I think this approach would work well as long as there are some features...

* Allow the user to add translations easily (like `TKUPT`)
* Hint/remind the user about the "short" version (like the suggestions window, but in this case it might be more useful)
* Add "short" and "long" translation version
* Edit and delete, of course.

With only `rofi` it might not be that easy.

**Comment by user202729** [comment]
Perhaps this can be used for things like punctuations too.

**Comment by sachac** [comment]
StenoTray has phrase suggestions. Is that something that might work in the meantime? My fork at https://github.com/sachac/StenoTray updates it for v4 config format.

**Reply by user202729** [reply]
That isn't really what I'm looking for (but the idea of guessing what the user want is similar to dictionary generation, which I (and others too) have been trying. There's https://github.com/openstenoproject/plover/discussions/1319#discussioncomment-748328 , but that is only based on orthography rules, and creates quite some false positives)

*by the way that Makefile (`make run`) doesn't work (`StenoTray.class` remains uncompiled) if there's no existing `StenoTray.class` file (because of how glob works in Makefile); besides that precompiled jar file is only compatible with new versions, and the default folder on (GNU/)Linux systems nowadays are .local/share/plover instead of .config/plover, and dictionary paths that contain `~` are not recognized*

**Comment by user202729** [comment]
First version.

https://github.com/user202729/plover-search-translation
